gpfit/fit.py

```diff
@@ -67,9 +65,17 @@ def fit(xdata, ydata, K, ftype="ISMA"):
     # Dimension of function (number of independent variables)
     d = int(xdata.shape[1])
 
-    with NamedVariables("fit"):
-        u = VectorVariable(d, "u")
-        w = Variable("w")
+    # begin fit data dict to generate fit constraint set
+    fitdata = {"ftype": ftype, "K": K, "d": d}
+
+    # find bounds of x
+    if d == 1:
+        fitdata["lb0"] = exp(min(xdata.reshape(xdata.size)))
+        fitdata["ub0"] = exp(max(xdata.reshape(xdata.size)))
+    else:
+        for i in range(d):
+            fitdata["lb%d" % i] = exp(min(xdata.T[i]))
```

Comment:
I'll have to dig into this to make sure, but this doesn't feel like the right abstraction level for fitdata...I feel like either just passing alpha/B/A and ftype, or passing a few lists might reduce the total lines of code between the two files.